### ISSUE TITLE
doom-themes-ext-visual-bell.el:0:0: error: cl-assertion-failed: ((> (point) output-start))

### ISSUE THREAD

**Issue by telotortium** [hlissner/emacs-doom-themes#655]
## What happened?

Compilation failed with the following message:

```
    x There was an unexpected error
      Message: error
      Data: (error . "doom-themes-ext-visual-bell.el:0:0: error: cl-assertion-failed: ((> (point) output-start))")
      Backtrace:
        (error "%s:0:0: error: %s: %s" "doom-themes-ext-visual-bell.el" cl-asserti
        (autoload-generate-file-autoloads "doom-themes-ext-visual-bell.el" nil "/U
        (make-directory-autoloads "/Users/rmirelan/.doom.d/doom.emacs.d/.local/str
        (if (fboundp 'make-directory-autoloads) (make-directory-autoloads (straigh
        (let ((find-file-hook nil) (write-file-functions nil) (debug-on-error nil)
        (if (file-exists-p generated-autoload-file) nil (let ((find-file-hook nil)
        (let ((generated-autoload-file (straight--autoloads-file package)) (backup
        (let* ((--cl-rest-- recipe) (package (car (cdr (plist-member --cl-rest-- '
        (straight--build-autoloads (:type git :flavor melpa :files (:defaults "the
        (run-hook-with-args straight--build-autoloads (:type git :flavor melpa :fi
```

## System Information
+ Emacs version: 28.0.50
+ Emacs distro: Doom Emacs
+ Operating system: MacOS
+ Installed commit of doom-themes: 9e2680b

**Comment by telotortium** [comment]
Strangely, when I run with the `-d` flag to the `doom` command, I don't see this error.

**Comment by telotortium** [comment]
By the way, I'm using nativecomp.

At this point, I had just upgraded my Emacs binary, and was trying to upgrade Doom as well. Even though the version remained the same at 28.0.50, perhaps

Hmm, I was able to fix this and https://github.com/hlissner/doom-emacs/issues/5459 by

1. Running `rm -rf ~/doom.emacs.dir/.local/straight/build*`.
2. Running `doom sync -p`.

I also saw the following when running the latter command:

```
      > Purging old native bytecode...
        - Deleting "28.0.50-43e5894e"
        ✓ Purged 1 directory(ies)
```

So maybe it's worth automatically purging old build artifacts and rebuilding everything (or at least prompting to do so) whenever Doom detects that `comp-native-version-dir` has changed.
